pkg/odo/cli/component/delete.go

```diff
@@ -210,6 +181,44 @@ func (do *DeleteOptions) Run() (err error) {
 	return
 }
 
+// Run has the logic to perform the required actions as part of command for devfiles
+func (do *DeleteOptions) DevFileRun() (err error) {
+	// devfile delete
+	if do.componentForceDeleteFlag || ui.Proceed(fmt.Sprintf("Are you sure you want to delete the devfile component: %s?", do.EnvSpecificInfo.GetName())) {
+		err = do.DevfileComponentDelete()
+		if err != nil {
+			log.Errorf("error occurred while deleting component, cause: %v", err)
+		}
+	} else {
+		log.Error("Aborting deletion of component")
+	}
+
+	if do.componentDeleteAllFlag {
+		if do.componentForceDeleteFlag || ui.Proceed(fmt.Sprintf("Are you sure you want to delete env folder?")) {
+			if !do.EnvSpecificInfo.EnvInfoFileExists() {
+				return fmt.Errorf("env folder doesn't exist for the component")
+			}
+			if err = util.DeleteIndexFile(filepath.Dir(do.devfilePath)); err != nil {
+				return err
+			}
+
+			err = do.EnvSpecificInfo.DeleteEnvInfoFile()
+			if err != nil {
+				return err
+			}
+			err = do.EnvSpecificInfo.DeleteEnvDirIfEmpty()
+			if err != nil {
```

Comment:
IMO, we should remove `.odo folder` also for `odo delete --all`, as `--all` signifies to delete all odo related data.

Reply:
Fixed